src/mame/konami/viper.cpp

```diff
@@ -776,121 +775,346 @@ void viper_state::pci_config_data_w(uint64_t data)
+		}),
+		NAME([this] (offs_t offset, u8 data) {
+			m_epic.i2c_sr = data;
+			LOGI2C("I2CSR %02x\n", data);
+		})
+	);
+	map(0x03010, 0x03010).rw(FUNC(viper_state::epic_i2cdr_r), FUNC(viper_state::epic_i2cdr_w));
+//  map(0x04000, 0x3ffff) <reserved>
+
+//  map(0x40000, 0x7ffff) EPIC
+	map(0x41030, 0x41033).lw32(
+		NAME([this] (offs_t offset, u32 data, u32 mem_mask) {
+			data = swapendian_int32(data);
+			m_epic.eicr = data;
+			LOG("EICR %08x\n", data);
+			if (BIT(data, 27))
+				throw emu_fatalerror("EPIC: serial interrupts mode not implemented\n");
+		})
+	);
+	map(0x41080, 0x41083).lr32(
+		NAME([this] (offs_t offset) {
+			if (!machine().side_effects_disabled())
+				LOG("EVI read\n");
+			// step = 1, device_id = 0, vendor_id = 0
+			return swapendian_int32(0x00010000);
+		})
+	);
+	map(0x410e0, 0x410e0).lw8(
+		NAME([this] (offs_t offset, u8 data) {
+			m_epic.svr = data;
+			LOGIRQ("SVR %02x\n", data);
+		})
+	);
+
+	map(0x41110, 0x41113).select(0xc0).lw32(
+		NAME([this] (offs_t offset, u32 data, u32 mem_mask) {
+			int timer_num = offset >> 4;
+			data = swapendian_int32(data);
+
+			m_epic.global_timer[timer_num].enable = (data & 0x80000000) ? 0 : 1;
+			m_epic.global_timer[timer_num].base_count = data & 0x7fffffff;
+
+			if (m_epic.global_timer[timer_num].enable && m_epic.global_timer[timer_num].base_count > 0)
+			{
+				attotime timer_duration =  attotime::from_hz((SDRAM_CLOCK / 8) / m_epic.global_timer[timer_num].base_count);
+				m_epic.global_timer[timer_num].timer->adjust(timer_duration, timer_num);
 
-	return nullptr;
+				LOGTIMER("EPIC GTIMER%d: next in %s\n", timer_num, (timer_duration / 8).as_string() );
+			}
+			else
+			{
+				m_epic.global_timer[timer_num].timer->reset();
+			}
+		})
+	);
+
+	map(0x41120, 0x41123).select(0xc0).lrw32(
+		NAME([this] (offs_t offset) {
+			u32 ret = 0;
+			int timer_num = offset >> 4;
+
+			ret |= m_epic.irq[MPC8240_GTIMER0_IRQ + timer_num].mask ? 0x80000000 : 0;
+			ret |= m_epic.irq[MPC8240_GTIMER0_IRQ + timer_num].priority << 16;
+			ret |= m_epic.irq[MPC8240_GTIMER0_IRQ + timer_num].vector;
+			ret |= m_epic.irq[MPC8240_GTIMER0_IRQ + timer_num].active ? 0x40000000 : 0;
+			return swapendian_int32(ret);
+		}),
+		NAME([this] (offs_t offset, u32 data, u32 mem_mask) {
+			int timer_num = offset >> 4;
+
+			data = swapendian_int32(data);
+
+			m_epic.irq[MPC8240_GTIMER0_IRQ + timer_num].mask = (data & 0x80000000) ? 1 : 0;
+			m_epic.irq[MPC8240_GTIMER0_IRQ + timer_num].priority = (data >> 16) & 0xf;
+			m_epic.irq[MPC8240_GTIMER0_IRQ + timer_num].vector = data & 0xff;
+
+			LOGIRQ("GTVPR%d %08x\n", timer_num, data);
+
+			if (!machine().side_effects_disabled())
+				epic_update_interrupts();
+		})
+	);
+	map(0x41130, 0x41130).select(0xc0).lw8(
+		NAME([this] (offs_t offset, u8 data) {
+			int timer_num = offset >> 4;
+
+			m_epic.irq[MPC8240_GTIMER0_IRQ + timer_num].destination = data & 0x1;
+
+			if (data)
+				throw emu_fatalerror("GTDR%d in P1 mode", timer_num, data & 1);
```

Comment:
I converted all of these because they clearly don't make sense in Konami Viper context and they are actively unused (they dispatch irq to a different processor if enabled).